pkg/storage/ec/client_test.go

```diff
@@ -166,17 +169,24 @@ TestLoop:
 
 		m := make(map[*pb.Node]client.PSClient, len(tt.nodes))
 		for _, n := range tt.nodes {
-			if !tt.badInput {
-				derivedID, err := id.Derive([]byte(n.GetId()))
-				if !assert.NoError(t, err, errTag) {
-					continue TestLoop
+			if n != nil {
+				if !tt.badInput {
```

Comment:
It would be better to combine the above to if-clauses into a single one (`if n != nil && !tt.badInput`) and avoid another level of nesting.

We can save even more nesting with:
```go
for _, n := range tt.nodes {
	if n == nil || tt.badInput {
		continue
	}
	...
}
```